option_download.py

```diff
@@ -0,0 +1,221 @@
+                    history = self.downloader.get_history(call_contract, start=start_date, end=end_date, interval="1m")
+                    if(history == False):
+                        log.e("unable to download {} history".format(call_contract))
+                        continue
+                    log.v("downloaded {} call contract data".format(call_contract))
+                    log.v("attempting to upload {} call contract".format(call_contract))
+                    self.importer.from_contents(history)
+                    log.v("uploaded {} call contract".format(call_contract))
+
+                log.i("downloading trade history of puts")
+                for put_contract in self.downloader.get_chain_contracts(ticker, expiration, "puts"):
+                    log.v("working on contract {}".format(put_contract))
+                    history = self.downloader.get_history(put_contract, start=start_date, end=end_date, interval="1m")
+                    if(history == False):
+                        log.e("unable to download {} history".format(put_contract))
+                        continue
+                    log.v("downloaded {} put contract data".format(call_contract))
+                    log.v("attempting to upload {} put contract".format(call_contract))
+                    self.importer.from_contents(history)
+                    log.v("uploaded {} put contract".format(call_contract))
+
+            log.i("finished ticker {}".format(ticker))
+
+def retrieve_ticker_list(doc_path: Path) -> List[str]:
+    '''
+    Grabs all tickers from the properly formatted doc_path file.\n
+
+    Returns:\n
+        A list of str, names of tickers.\n
+    '''
+    doc = json.load(doc_path.open("r"))
+    return [ticker['ticker'] for ticker in doc['tickers']]
+
+
+def list_tickers_file(ticker_list_folder: Path) -> Path:
+    '''
+    List json files from the docs folder where to find the ticker list.
+
+    Returns:
+        Path to the selected ticker list file.
+    '''
+    docs_glob = ticker_list_folder.glob('*.json')
+    return [x.name.replace('.json', '') for x in docs_glob if x.is_file()]
+
+
+def get_seven_days_ago() -> date:
+    '''
+    Calculates when is 7 days ago.\n
+
+    Returns:\n
+        The date of 7 days ago.\n
+    '''
+    seven_days_delta = timedelta(days=7)
+    seven_days_ago = datetime.now() - seven_days_delta
+    return date(seven_days_ago.year, seven_days_ago.month, seven_days_ago.day)
+
+
+def print_error_msg(msg: str = None):
+    if msg is None:
+        log.e("option_download.py -p <provider: {}> -f <ticker file: {}>".format(
+            list(DOWNLOADERS.keys()),
+            list_tickers_file(TICKER_LISTS_FOLDER)
+        )
+        )
+    else:
+        log.e("{}: option_download.py -p <provider: {}> -f <ticker file: {}>".format(
+            msg,
+            list(DOWNLOADERS.keys()),
+            list_tickers_file(TICKER_LISTS_FOLDER)
+        )
+        )
+
+
+if __name__ == "__main__":
```

Comment:
Codacy found an issue: [If 145 is too complex (16) (MC0001)](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5905963)